src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -12939,9 +12939,17 @@ private bool IsPossibleCollectionElement()
 
         private CollectionElementSyntax ParseCollectionElement()
         {
-            return IsAtDotDotToken()
-                ? _syntaxFactory.SpreadElement(this.EatDotDotToken(), this.ParseExpressionCore())
-                : _syntaxFactory.ExpressionElement(this.ParseExpressionCore());
+            if (this.CurrentToken.ContextualKind == SyntaxKind.WithKeyword &&
+                this.PeekToken(1).Kind == SyntaxKind.OpenParenToken &&
+                IsFeatureEnabled(MessageID.IDS_FeatureCollectionExpressionArguments))
+            {
+                return _syntaxFactory.WithElement(this.EatContextualToken(SyntaxKind.WithKeyword), this.ParseParenthesizedArgumentList());
+            }
+
+            if (this.IsAtDotDotToken())
+                return _syntaxFactory.SpreadElement(this.EatDotDotToken(), this.ParseExpressionCore());
```

Comment:
nit: would prefer braces since expression isn't trivial, but technically .NET guidelines allow

Reply:
this is very intentional.  we originally required braces, but then mandated that we have to follow .net rules instead.  so this is the fallout :)